src/components/browse/dataset/datasetPagination.tsx

```diff
@@ -1,30 +1,53 @@
-import React from 'react';
-import { Pagination } from 'react-bootstrap';
+import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
+import ReactPaginate from 'react-paginate'
+import { searchResponseModel } from "../../../models/dataset";
+import { getDatasetsSearchResp } from "../../../api/browse";
+import { facetFilterModel } from '../../../models/facets'
 
 interface dataSetPaginationProps {
-  dsCount : number
-  active : number
-  setActive: React.Dispatch<React.SetStateAction<number>>
+  dsCount: number;
+  setSearchResp: Dispatch<SetStateAction<searchResponseModel | null>>;
+  searchKeyword: string;
+  limit: number;
+  setLimit: any;
+  filterDict: facetFilterModel[];
 };
 
 const DatasetPagination = (props: dataSetPaginationProps) => {
+  const [pageCount, setpageCount] = useState(0);
 
-  const PaginationGenerator = (active : number = 1) => {
-    var items = []
-    for (let i = 1; i <= props.dsCount/10; i++) {
-      items.push(<Pagination.Item key={i} active={i === active} onClick={()=> props.setActive(i)}>{i}</Pagination.Item>)
-    }
+  useEffect(() => {
+    setpageCount(Math.ceil(props.dsCount / props.limit))
+  }, [props.dsCount, props.limit]);
 
-  return items
-}
+  const handlePageClick = async (data: any) => {
+    let skip = (data.selected) * props.limit;
+    getDatasetsSearchResp(props.setSearchResp, props.filterDict, props.searchKeyword, skip, props.limit)
+  };
 
   return (
-    <div>
-        <Pagination>
-            {PaginationGenerator(props.active)}
-        </Pagination>
+    <div className="container">
+      <ReactPaginate
+        previousLabel={"<"}
+        nextLabel={">"}
+        pageCount={pageCount}
+        marginPagesDisplayed={4}
```

Comment:
Should be 3 I think

Reply:
This would be later determined by a variable or from config params.

Reply:
How so? No need to define the look of the pagination by config.